dataset/dataset.cpp

```diff
@@ -152,16 +152,24 @@ Dataset Dataset::slice(const Slice s) const {
 }
 
 Dataset &Dataset::setSlice(const Slice s, const Dataset &data) {
-  // TODO Need dry-run mechanism here?
-  for (const auto &item : data)
-    (*this)[item.name()].setSlice(s, item);
+  // Validate slice on all items as a dry-run
+  expect::coordsAreSuperset(slice(s), data);
+  for (const auto &[name, item] : m_data)
+    item.validateSlice(s, data[name]);
```

Comment:
```suggestion
    item.validateSlice(s, data.m_data[name]);
```
or else you will have the coords of the RHS included?